QUARKUS-6258.md

```diff
@@ -38,14 +49,19 @@ Given the above I do not recommend to (re)use this repository for product valida
 There are integration tests modules in the extension, but many of them use mocking or don't contain tests at all.
 According to Mario, this is intentional, since they consider upstream lang4chain tests to be enough (but we do not!)
 Ollama modules can be started on the laptop in dev mode. Jlama ones can be run as well.
-The features we want to cover are being used in integration-tests modules for OpenAI (Chats, Chat Memory, user/system messages, Function calling/Tools, Guardrails), RAG,. There is no coverage for WebSockets (only in codestarts) and for ChatModel (only in mocktests).
+The features we want to cover are being used in integration-tests modules for OpenAI (Chats, Chat Memory, user/system messages, Function calling/Tools, Guardrails) and RAG. There is no coverage for WebSockets (only in codestarts) and for ChatModel (only in mocktests).
 
-### Impact on test suite
-We will focus on updating and running upstream tests from the Quarkiverse extensions against productised binaries.
+## Impact on test suite
+We will focus on updating and running upstream tests from the Quarkiverse extensions against productised binaries. The most promising targets are tests in `intergration-tests` folder (to be run on their own) or `samples` folder (to be run as an external app inside Quarkus QE TS). The repository of the extension should be updated so these modules can be used with provided RHBQ platform (currently they have the platform version hardcoded)
 
 Primary focus will be on RHEL-based testing to ensure functionality of proposed features against productized bits. For the OpenShift side, we will explore the possibility of using the `@GitRepositoryQuarkusApplication` approach.
 
-As we aim for tech preview level for 3.27, we won't be focusing on full platform matrix coverage. We will be gaining experience to expand the coverage and platforms in the upcoming releases.d testing.
+As we aim for tech preview level for 3.27, we won't be focusing on full platform matrix coverage. We will be gaining experience to expand the coverage and platforms in the upcoming releases.
+
+## Impact on test suites for Full Support
+Since Quarkus AI may be moved to full support in its first release (Quarkus 3.27), we need to have a full support on all (see below) platforms. This means, that all tests should be in Quarkus Test suite, to guarantee running an all supported platforms.
+Additionally, we should add verification for this extension to Marete and startstop tests.
+Note: Requirement about "all platforms" (without further details) came from the PM. Quarkus QE runs tests for RHEL, Openshift and Windows, so this is the platforms we will cover Quarkus AI on. While Quarkus is also supported on IBM Z& IBM P, it is out of scope for Quarkus QE team (there is a separate IBM team, which handles it). 
```

Comment:
I'm not really sure what this sentence means, or why is it important for test plan.

But if an extension or feature is declared to be fully supported, it must work on all the certified platforms. That's the definition. That's not what the reviewed sentence says, though.

Reply:
Yes, I requested to explain what "all supported platforms" mean. Is it just me who see a difference between "all supported platforms" meaning "platforms that we will support for AI" and all supported platforms" meaning "all platforms we support for RHBQ"? You are going to support Windows and OpenShift and Aarch64, cool. You can just tweak that text to provide clear meaning so that it can be verified. e.g.

```
[all supported platforms](https://access.redhat.com/articles/4966181)
```

is clear to me